UPGRADE-4.2.md

```diff
@@ -125,6 +125,26 @@ HttpKernel
 FrameworkBundle
 ---------------
 
+ * The `allow_no_handler` middleware has been removed. Use `framework.messenger.[bus].default_middleware` instead:
```

Comment:
You say the handler is removed, but below is „allow_no_handlers“... it is a bit confusing

Reply:
Actually the **middleware** is removed in favor of a flag in the `HandleMessageMiddleware`. I believe this information isn't really useful from the config POV so I omit it and just talk about setting the `framework.messenger.[bus].default_middleware` config option to `allow_no_handlers` to set this flag.
How can I improve this?

Reply:
if it's too hard to explain.. i suggested https://github.com/symfony/symfony/pull/28945#discussion_r227114703 :angel:

Reply:
@ro0NL : At least, I agree`allow_no_handlers` at the bus node level would be self-explanatory.

Reply:
it also brings more flexibility, i.e. keep logging but disable other defaults.

Being able to control it per case, IMHO would be more straightforward. Currently it's a bit all or nothing, and you need to know what default_middleware=true actually means.

Reply:
That's why I was never really convinced by the default middleware config over explicit middleware stack configuration.

Reply:
but then we cant provide sensible defaults (including default ordering).. so having a node per default + custom middleware stack seems like best of both.

So the only "weird" case is, if you want re-order defaults

```yaml
bus:
  logging: false # log before
  middleware: [custom, logging] # log after
```

Reply:
That's still requires knowledge about what the defaults are and their order.

We could add a `framework.messenger.default_middleware` with `before` and `after` entries (or a `[middleware]` placeholder for specific bus middleware), with same format as middlewares & default values set in the `Configuration` allowing these to be dumped on `debug:config` and changed in userland so anyone can provide their own defaults, but not sure it is worth.

```yml
framework:
    messenger:
        default_middleware: ~
        # Which by default is the same as:
        # default_middleware:
        #   before:
        #     - logging
        #   after:
        #     - route_messages
        #     - call_message_handler
        bus:
            command:
                middleware:
                    - doctrine_transaction
            query:
                default_middleware: false
                middleware:
                    - cache
                    - call_message_handler
            events:
                allow_no_handlers: true
```

But defaults may also be a PITA if one day we need to deprecate a middleware part of defaults in favor of another one. We cannot either add more or remove one in the future.

IMHO, explicit is always better.

Reply:
maybe you're right :+1: the defaults are not super complex to declare anyway...